lib/msf/core/module/external.rb

```diff
@@ -152,6 +152,11 @@ def handle_credential_login(data, mod)
     credential_data[:private_type] = :password
   end
 
+  if data.has_key?(:domain)
+    credential_data[:public_data] = data['domain']
+    credential_data[:public_type] = :realm
```

Comment:
Instead of changing the credential type, I think this just needs to add a realm to the credential data:
```suggestion
    credential_data[:realm] = data['domain']
```

Reply:
I just tried this, but it doesn't seem to work either. I'm wondering if the `create_credential` function called at the end isn't handling the realm after it's added or something or if I'm still passing it in incorrectly. I've tried adding it with :realm_key as well because i saw that as I was searching but it didn't seem to work either